From 8e7797e90104e47a50611cf62e246ee9de67ad95 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:45:15 +0530
Subject: [PATCH] system udev (rebased for 43.6)

Photon does not define distro_redhat. Apply the same udev_t rules as the
RHEL block under distro_photon (hugetlbfs, ptys, netutils, unconfined).

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/system/udev.te | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/policy/modules/system/udev.te b/policy/modules/system/udev.te
index a0f302b..897d7cd 100644
--- a/policy/modules/system/udev.te
+++ b/policy/modules/system/udev.te
@@ -232,6 +232,19 @@ ifdef(`distro_gentoo',`
 	init_getattr_script_status_files(udev_t)
 ')
 
+ifdef(`distro_photon',`
+	fs_manage_hugetlbfs_dirs(udev_t)
+
+	term_use_generic_ptys(udev_t)
+
+	# for arping used for static IP addresses on PCMCIA ethernet
+	netutils_domtrans(udev_t)
+
+	optional_policy(`
+		unconfined_domain(udev_t)
+	')
+')
+
 ifdef(`distro_redhat',`
 	fs_manage_hugetlbfs_dirs(udev_t)
 
